qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/IqtFunctionModel.cpp

```diff
@@ -557,7 +557,7 @@ void IqtFunctionModel::estimateExpParameters() {
     auto const &x = m_estimationData[i].x;
     auto const &y = m_estimationData[i].y;
     auto lifeTime = (x[1] - x[0]) / (log(y[0]) - log(y[1]));
-    if (lifeTime <= 0)
+    if (lifeTime <= 0 || isnan(lifeTime))
```

Comment:
since any comparison between a NaN and a double should always be false we can simplify this instead with this.

```suggestion
    if (!lifeTime > 0)
```

Reply:
I think your proposal isn't correct.
I would write it like this:
   if (!(lifeTime > 0))
which would give the right results (at least for me)